docs/current_docs/developer-guide/go/742930-services.mdx

```diff
@@ -0,0 +1,227 @@
+
+```shell
+dagger call http-service up
+```
+
+By default, each service port maps to the same port on the host - in this case, port 8080. The service can then be accessed by clients on the host. Here's an example:
+
+```shell
+curl localhost:8080
+```
+The result will be:
+
+```shell
+Hello, world!
+```
+
+To specify a different mapping, use the additional `--ports` argument to `dagger call ... up` with a list of host/service port mappings. Here's an example, which exposes the service on host port 9000:
+
+```shell
+dagger call http-service up --ports 9000:8080
+```
+
+:::note
+To bind ports randomly, use the `--random` argument.
+:::
+
+## Expose host services to functions
+
+Dagger Functions can also receive host services as function arguments of type `Service`, in the form `tcp://<host>:<port>`. This enables client containers in Dagger Functions to communicate with services running on the host.
+
+:::note
+This implies that a service is already listening on a port on the host, out-of-band of Dagger.
+:::
+
+Here is an example of how a container running in a Dagger Function can access and query a MariaDB database service (bound using the alias `db`) running on the host.
+
+```go file=./snippets/services/expose-host-services-to-dagger/main.go
+```
+
+Before calling this Dagger Function, use the following command to start a MariaDB database service on the host:
+
+```shell
+docker run --rm --detach -p 3306:3306 --name my-mariadb --env MARIADB_ROOT_PASSWORD=secret  mariadb:10.11.2
+```
+
+Here is an example call for this Dagger Function:
+
+```shell
+dagger call user-list --svc=tcp://localhost:3306
+```
+
+The result will be:
+
+```shell
+Host    User
+%       root
+localhost       mariadb.sys
+localhost       root
+```
+
+## Persist service state
+
+Dagger cancels each service run after a 10 second grace period to avoid frequent restarts. To avoid relying on the grace period, use a cache volume to persist a service's data, as in the following example:
+
+```go file=./snippets/services/persist-service-state/main.go
+```
+
+This function uses Redis's SAVE command to ensure data is synced. By default, Redis flushes data to disk periodically.
+
+Here is an example call for this Dagger Function:
+
+```shell
+dagger call redis-service
+```
+
+The result will be:
+
+```shell
+abc
+```
+
+## Start and stop services
+
+Services are designed to be expressed as a Directed Acyclic Graph (DAG) with explicit bindings allowing services to be started lazily, just like every other DAG node. But sometimes, you may need to explicitly manage the lifecycle in a Dagger Function.
+
+For example, this may be needed if the application in the service has certain behavior on shutdown (such as flushing data) that needs careful coordination with the rest of your logic.
+
+Here's an example which demonstrates explicitly starting a Docker daemon for use in a test suite:
```

Comment:
This part just needs to be tweaked now that it's not using the Docker example, and instead builds on examples from prior sections:

```suggestion
The following example explicitly starts the Redis service and stops it at the end, ensuring the 10 second grace period doesn't get in the way, without the need for a persistent cache volume:
```

Reply:
Added